backend/core/src/cron/cron.service.ts

```diff
@@ -1,10 +1,45 @@
 import { Injectable } from "@nestjs/common"
 import { Cron, CronExpression } from "@nestjs/schedule"
+import { UserListQueryParams } from "../auth/dto/user-list-query-params"
+import { UserService } from "../auth/services/user.service"
+import { AuthContext } from "../auth/types/auth-context"
+import { Compare } from "../shared/dto/filter.dto"
+import { EmailService } from "../shared/email/email.service"
+import { ListingsService } from "../listings/listings.service"
 
 @Injectable()
 export class CronService {
-  @Cron(CronExpression.EVERY_WEEK)
-  handleCron() {
-    // TODO: add a cron task here.
+  constructor(
+    private readonly emailService: EmailService,
+    private readonly listingsService: ListingsService,
+    private readonly userService: UserService
+  ) {}
+
+  @Cron(CronExpression.EVERY_1ST_DAY_OF_MONTH_AT_NOON)
+  async handleCron() {
+    if (!process.env.SEND_NOTIFICATIONS_FOR_UPDATE_LISTINGS_REMINDER) {
+      return
+    }
+
+    const userQueryParams: UserListQueryParams = {
+      filter: [
+        {
+          $comparison: Compare["EQUALS"],
+          isPartner: true,
+        },
+      ],
+    }
+    const users = await this.userService.list(userQueryParams, new AuthContext())
+    const allListings = await this.listingsService.list({})
+
+    // For each listing, check whether the listed leasing agents are in the list of partner users.
+    // If the leasing agent is a partner and has their email notifications turned on, send the reminder email.
+    for (const listing of allListings.items) {
```

Comment:
Since we're sending 1 email per listing couldn't we do most of this logic in a single SQL query? I'm a little rusty on db queries so maybe this is less efficient, but what if we do a join between the listings, user, and user preferences tables and filter to return only listings that have a leasing agent with email notifications enabled. Then here we would just have to loop over each listing and get the leasing agent email from that joined table?

Reply:
I also considered doing a single SQL query but the way that the code is set up we only really have functions that query a single table - like its easy to query listings separately from users. I can look into it more but for now I think this is the simplest way to implement, maybe I can open a bug here for exygy to make it more efficient in the future if that sounds good to you